### PR TITLE
Feature - 4228 - Skill picker updates

### LINKED ISSUE #4228: Changes to Skill Picker
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Screenshot: Prototype
[![a screenshot of ...][image]][image-link]
<sup>[Click screenshot to visit Figma.][image-link]</sup>
Screen Shot 2022-10-06 at 3 25 13 PM
Screen Shot 2022-10-06 at 3 26 08 PM

## Acceptance Criteria
- [ ] Remove tabs.
- [ ] Keyword search box goes first, followed by skill family dropdown, then list of skills.
- [ ] Skill family filtering in one multiselect, with occupational and transferable skills grouped. Selecting multiple skill families returns a union of filters. Having no skill families selected shows all skills (doesn't filter out skills).
    - If grouping in multiselect is complicated, could be turned into two multiselects. 

   [image]: https://user-images.githubusercontent.com/305339/166570935-6425a615-82c7-4b2f-9840-5c7d698fa2ab.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by esizer**
Resolves #4228 and #4230 

## Summary

This overhauls the `<SkillPicker />` to replace the tab interface with a search bar and select field for skill families.

## Details

The new `<SkillPicker />` has some changes that should also improve accessibility.

- It announces the number of skills found once you search
- Tracks `aria-expanded` on the "See definition" toggle
- Improves context of the "See definition" and "Add skill" buttons
- Announces the definition once opened (not 100% sure if this is a pro or con yet, thoughts?)

## Notes for Designers

Since I was working from a wireframe, the styling is all done by me and I'd love some feedback to improve it!

## TO DO

- [x] Support `optgroup` in `MultiSelectV2`
- [x] Update `e2e` tests

## Screenshot

Screen Shot 2022-10-07 at 12 18 08 PM

## Testing

> **Warning**
> If you are doing a11y testing in storybook on a mac, there is a known bug that VoiceOVer double announces `aria-live` areas. This means, while searching, you will hear "Found X skills" twice every time. This only occurs in storybook and not on the actual site.

### Storybook

This is fully functional in story book and you can do testing there but, it is recommended to do the testing in context and confirm none of them break.

### Locations

This component can be found the following locations to be tested:

- `/seach`
- `/users/{userId}/profile/experiences/{experiencType}/create`
- `/users/{userId}/profile/experiences/{experiencType}/{experienceId}/edit`
- `/admin/pools/{poolId}/edit`

## Steps

1. [ ] Build both projects `npm run production --workspaces`
2. [ ] Navigate to one of the locations listed in the "Locations" section
3. [ ] Fill out a search term and confirm expected results appear
4. [ ] Select some skill families and confirm expected results appear
5. [ ] Combine steps 3-4 confirm expected results appear
6. [ ] Select "See definition" and confirm the definition appears
7. [ ] Confirm you can hide the definition
8. [ ] Select "Add skill" and confirm it appears
9. [ ] Confirm you can remove a skill with the "Remove skill" and `<Chip />` x icon
10. [ ] Confirm you can scroll the area with skills
11. [ ] Bonus: Do steps 3-10 with a screen reader and just your keyboard 😄

### CODE REVIEW

**Comment by patcon** [comment]
real quick before i sign-off for night: went to scope stories for selectfieldv2 and multiselectfieldv2, and didn't see any. (will do real review next week)

**Comment by esizer** [comment]
> real quick before i sign-off for night: went to scope stories for selectfieldv2 and multiselectfieldv2, and didn't see any. (will do real review next week)

Are you referring to stories showcasing the `optgroup`? Or are they missing entirely for you? I will add the `optgroup` stories, forgot about them after all the bug hunting 🐛 🏹

File: frontend/common/src/components/Pill/Pill.tsx [hunk 1]
```diff
@@ -52,23 +56,38 @@ const colorMap: Record<
...
 };
 
+const sizeMap: Record<PillSize, Record<string, string>> = {
+  sm: {
+    "data-h2-font-size": "base(0.6rem)",
+    "data-h2-padding": "base(x.1, x.25)",
+  },
+  md: {
+    "data-h2-padding": "base(x.25, x.75)",
+    "data-h2-font-size": "base(caption)",
```
Comment (tristan-orourke) [inline frontend/common/src/components/Pill/Pill.tsx:66]: Just to make it consistent with the other sizes 😅 
```suggestion
    "data-h2-font-size": "base(caption)",
    "data-h2-padding": "base(x.25, x.75)",
```

File: frontend/common/src/components/SkillPicker/SkillBlock.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            </Button>
+          </div>
+          <div>
+            {definition ? (
+              <Collapsible.Trigger asChild>
+                <Button
+                  color="primary"
+                  mode="inline"
+                  type="button"
+                  onClick={() => setIsOpen(!isOpen)}
```
Comment (tristan-orourke) [inline frontend/common/src/components/SkillPicker/SkillBlock.tsx:121]: When a useState setter uses its own value, I think its best practice to pass the current value as an argument:
https://reactjs.org/docs/hooks-reference.html#functional-updates

```suggestion
                  onClick={(currentIsOpen) => setIsOpen(!currentIsOpen)}
```

**Review by tristan-orourke — APPROVED** [review]
Everything looks great. Some interesting new utility components too, and updates to the Select input!

Just before merging, could you translate "Skill Families" and "Found N skills"? Everything else in the picker seems to be already translated.
image

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
I guess approval should also be contingent on all the tests passing :/

**Comment by esizer** [comment]
@tristan-orourke  Added French in [update copy en + fr](https://github.com/GCTC-NTGC/gc-digital-talent/pull/4242/commits/099655808da9d6c47834e8251988ce01cb1a07d2)

**Review by tristan-orourke — APPROVED** [review]
👏

**Pull request merged**